developer/src/common/web/utils/src/types/kps/kps-file-reader.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *
+ * Created by mcdurdin on 2024-10-16
+ *
+ * XML reader for .kps file format
+ */
+
+import { util } from '@keymanapp/common-types';
+import boxXmlArray = util.boxXmlArray;
+
+import { CommonTypesMessages } from "../../common-messages.js";
+import { CompilerCallbacks } from "../../compiler-interfaces.js";
+import { KeymanXMLReader } from "../../xml-utils.js";
+import { KpsPackage } from "./kps-file.js";
+
+export class KpsFileReader {
+  constructor(private callbacks: CompilerCallbacks) {
+
+  }
+
+  public read(file: Uint8Array): KpsPackage {
+    const data = new TextDecoder().decode(file);
+
+    const kpsPackage = (() => {
+        let a: KpsPackage;
+
+        try {
+          a = new KeymanXMLReader('kps')
```

Comment:
Maybe overkill, but thoughts on using `KeymanXMLType.kps` instead of `'kps'`?

On [kps-file-writer.ts](https://github.com/keymanapp/keyman/pull/12545/files#diff-5be57173af38b88136e1c6bb2197094cbef7456497b726b7995351d9b42fa5e7) too

Reply:
It's not needed because Typescript type-checks the string so invalid strings aren't accepted